File: src/deepview_canvas/widgets/canvas_duplicates/src/widget.ts
// For licensing see accompanying LICENSE file.
// Copyright (C) 2023 betterwithdata Inc. All Rights Reserved.

import { MODULE_NAME, MODULE_VERSION } from './version';
import {
  DOMWidgetModel,
  DOMWidgetView,
  ISerializers,
} from '@jupyter-widgets/base';

import {serializers, JupyterWidget } from '@betterwithdata/canvas_viz';

import CanvasDuplicates from './CanvasDuplicates.svelte';

export class CanvasDuplicatesModel extends DOMWidgetModel {
  defaults() {
    return {
      ...super.defaults(),
      _model_name: CanvasDuplicatesModel.model_name,
      _model_module: CanvasDuplicatesModel.model_module,
      _model_module_version: CanvasDuplicatesModel.model_module_version,
      _view_name: CanvasDuplicatesModel.view_name,
      _view_module: CanvasDuplicatesModel.view_module,
      _view_module_version: CanvasDuplicatesModel.view_module_version,
    };
  }
  static serializers: ISerializers = serializers;

  static model_name = 'CanvasDuplicatesModel';
  static model_module = MODULE_NAME;
  static model_module_version = MODULE_VERSION;
  static view_name = 'CanvasDuplicatesView'; // Set to null if no view
  static view_module = MODULE_NAME; // Set to null if no view
  static view_module_version = MODULE_VERSION;
}

export class CanvasDuplicatesView extends DOMWidgetView {
  render() {
    new JupyterWidget({
      target: this.el,
      props: {
        model: this.model,
        widget: CanvasDuplicates,
      },
    });
  }
}

File: src/deepview_canvas/widgets/canvas_duplicates/canvas_duplicates/widget.py
# For licensing see accompanying LICENSE file.
# Copyright (C) 2023 betterwithdata Inc. All Rights Reserved.

from pathlib import Path

import canvas_ux
from traitlets import Unicode

from ._frontend import module_name, module_version


class CanvasDuplicates(canvas_ux.CanvasWidget):
    _model_name = Unicode(
        'CanvasDuplicatesModel').tag(sync=True)
    _view_name = Unicode(
        'CanvasDuplicatesView').tag(sync=True)
    _model_module = Unicode(module_name).tag(sync=True)
    _model_module_version = Unicode(module_version).tag(sync=True)
    _view_module = Unicode(module_name).tag(sync=True)
    _view_module_version = Unicode(module_version).tag(sync=True)
    name = 'CanvasDuplicates'
    description = "A Canvas component for inspecting potential duplicates in a dataset"

    def __init__(self,
                 width: str = 'XXL',
                 height: str = 'M',
                 page: str = 'Duplicates',
                 **kwargs
                 ):
        """A Canvas component for inspecting potential duplicates in a dataset

        Parameters
        ----------
        width : str, optional
            By default "XXL".
        height : str, optional
            By default "M".
        page : str, optional
            Which page of the dashboard to show the widget on, by default "CanvasDuplicates"

        Returns
        -------
        CanvasDuplicates
        """
        super().__init__(**kwargs)
        self.width: str = width
        self.height: str = height
        self.page: str = page
        self.spec = canvas_ux.WidgetSpec(
            width=width,
            height=height,
            page=page,
            name=self.name,
            description=self.description
        )

    def js_path(self):
        app_files_path = Path(
            (Path(__file__).parent), 'standalone', 'widgets')
        js_path = Path(app_files_path, 'CanvasDuplicates.js')
        map_path = Path(
            app_files_path, 'CanvasDuplicates.js.map')
        return js_path, map_path